Src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs

```diff
@@ -231,7 +231,7 @@ protected virtual SyntaxNode TransformReducedNode(SyntaxNode reducedNode, Syntax
             }, cancellationToken);
             }
 
-            Task.WaitAll(simplifyTasks, cancellationToken);
```

Comment:
This whole discussion seems to suggest I _added_ the cancellationToken to the wait function. I did not. It was already there. I merely preserved prior behavior. The `WaitAll` method took a cancellationToken. This not only allowed the async method's `Task` to complete quickly upon cancellation regardless of the tasks it is waiting on, but it increased the odds that in that event an `OperationCanceledException` would be thrown instead of some other fault.
I'm happy to remove the `WithCancellationToken` extension method call if you would rather not have it though.